src/Dialect/ONNX/CMakeLists.txt

```diff
@@ -12,29 +12,23 @@ add_onnx_mlir_dialect_doc(onnx ONNXOps.td)
 add_onnx_mlir_rewriter(Rewrite)
 add_onnx_mlir_rewriter(Combine)
 
-# Header dependencies target for ONNXOps.hpp
-add_custom_target(OMONNXOpsInc
-        DEPENDS OMONNXOpsIncGen
-                OMResultTypeInferenceOpInterfaceIncGen
-                OMShapeInferenceOpInterfaceIncGen
-                OMHasOnnxSubgraphOpInterfaceIncGen
-                OMONNXRewriteIncGen
-                OMONNXCombineIncGen)
-
 add_onnx_mlir_library(OMONNXOps
-        ONNXOps.cpp
-        ONNXOpsHelper.cpp
-        ONNXShapeHelper.cpp
-        Rewrite.cpp
-        Combine.cpp
-        IndexExpr.cpp
-        IndexExprDetail.cpp)
+  ONNXOps.cpp
+  ONNXOpsHelper.cpp
+  ONNXShapeHelper.cpp
+  Rewrite.cpp
+  Combine.cpp
+  IndexExpr.cpp
+  IndexExprDetail.cpp
 
-add_dependencies(OMONNXOps OMONNXOpsInc)
-add_dependencies(OMONNXRewriteIncGen OMONNXOpsIncGen)
-add_dependencies(OMONNXCombineIncGen OMONNXOpsIncGen)
+  DEPENDS
+  OMHasOnnxSubgraphOpInterfaceIncGen
```

Comment:
It has a header dependency here, but really doesn't have a linkage dependency?  Usually, the corresponding libraries would be in LINK_LIBS PUBLIC

Reply:
That's what I expected also, but it is happy to build without the library, at least in my testing on Ubuntu and Windows. I'm curious to see if any of the automated builds will fail, but as long as they don't, I think this is fine.

Reply:
Usually BUILD_SHARED_LIBS=on is stricter about such dependencies..  It's possible it's not necessary if it's only using #defines and such.

Reply:
All the builds work correctly without the lib dependency, so I believe it is unnecessary.